base_user_role/models/user.py

```diff
@@ -44,8 +44,9 @@ def set_groups_from_roles(self):
                 lambda rec: rec.is_enabled)
             for role_line in role_lines:
                 role = role_line.role_id
-                group_ids.append(role.group_id.id)
-                group_ids.extend(role.implied_ids.ids)
+                if role:
+                    group_ids.append(role.group_id.id)
+                    group_ids.extend(role.implied_ids.ids)
             group_ids = list(set(group_ids))    # Remove duplicates IDs
```

Comment:
I think the method could be refactored to use more new-API magic.

Untested proposal (written directly in this comment):
```python
    @api.multi
    def set_groups_from_roles(self):
        """Set (replace) the groups following the roles defined on users.
        If no role is defined on the user, its groups are let untouched.
        """
        for user in self.filtered('role_line_ids'):
            role_lines = user.role_line_ids.filtered('is_enabled')
            groups = (
                role_lines.mapped('role_id.group_id') +
                role_lines.mapped('role_id.implied_ids')
            )
            super(ResUsers, user).write({
                'groups_id': [(6, 0, groups.ids)],
            })
        return True
```